File: 프로그래머스/lv1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.js
function gcd(a, b) {
    if (b === 0) return a;
    return gcd(b, a % b)
}

function solution(n, m) {
    const lcm = (a, b) => (a * b) / gcd(a, b);
    return [gcd(n, m), lcm(n, m)]
}

File: 프로그래머스/lv1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.py
def gcd(a, b):
    if b == 0: return a;
    return gcd(b, a % b)

def lcm(a, b):
    return a * b / gcd(a, b);

def solution(n, m):
    return [gcd(n, m), lcm(n, m)]